code/datums/components/toggle_coat.dm

```diff
@@ -0,0 +1,47 @@
+/datum/component/toggle_coat
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/buttoned = null
+	var/coat_style = null
+	var/obj/item/clothing/suit/suit = null
+	var/obj/ability_button/coat_toggle/toggle = new
+
+/datum/component/toggle_coat/Initialize(buttoned = null, coat_style = null)
```

Comment:
```suggestion
/datum/component/toggle_coat/Initialize(buttoned, coat_style)
```

Setting default args to `null` doesn't do anything since all args are null by default. BYOND lets you call a proc with any number of arguments, so calling this without passing any arguments (or with arguments of `[null, null]` if you're a weirdo) already keeps these null.